File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/KeyboardEntity.java
package dev.ragnarok.fenrir.db.model.entity;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class KeyboardEntity {
    @SerializedName("one_time")
    private boolean one_time;
    @SerializedName("inline")
    private boolean inline;
    @SerializedName("author_id")
    private int author_id;
    @SerializedName("buttons")
    private List<List<ButtonEntity>> buttons;

    public boolean getOne_time() {
        return one_time;
    }

    public KeyboardEntity setOne_time(boolean one_time) {
        this.one_time = one_time;
        return this;
    }

    public boolean getInline() {
        return inline;
    }

    public KeyboardEntity setInline(boolean inline) {
        this.inline = inline;
        return this;
    }

    public int getAuthor_id() {
        return author_id;
    }

    public KeyboardEntity setAuthor_id(int author_id) {
        this.author_id = author_id;
        return this;
    }

    public List<List<ButtonEntity>> getButtons() {
        return buttons;
    }

    public KeyboardEntity setButtons(List<List<ButtonEntity>> buttons) {
        this.buttons = buttons;
        return this;
    }

    public static class ButtonEntity {
        @SerializedName("color")
        private String color;
        @SerializedName("type")
        private String type;
        @SerializedName("label")
        private String label;
        @SerializedName("link")
        private String link;
        @SerializedName("payload")
        private String payload;

        public String getColor() {
            return color;
        }

        public ButtonEntity setColor(String color) {
            this.color = color;
            return this;
        }

        public String getType() {
            return type;
        }

        public ButtonEntity setType(String type) {
            this.type = type;
            return this;
        }

        public String getLabel() {
            return label;
        }

        public ButtonEntity setLabel(String label) {
            this.label = label;
            return this;
        }

        public String getLink() {
            return link;
        }

        public ButtonEntity setLink(String link) {
            this.link = link;
            return this;
        }

        public String getPayload() {
            return payload;
        }

        public ButtonEntity setPayload(String payload) {
            this.payload = payload;
            return this;
        }
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/KeyboardEntity.kt
package dev.ragnarok.fenrir.db.model.entity

import androidx.annotation.Keep
import kotlinx.serialization.Serializable

@Keep
@Serializable
class KeyboardEntity {
    var one_time = false
        private set
    var inline = false
        private set

    var author_id = 0L
        private set

    var buttons: List<List<ButtonEntity>>? = null
        private set

    fun setOne_time(one_time: Boolean): KeyboardEntity {
        this.one_time = one_time
        return this
    }

    fun setInline(inline: Boolean): KeyboardEntity {
        this.inline = inline
        return this
    }

    fun setAuthor_id(author_id: Long): KeyboardEntity {
        this.author_id = author_id
        return this
    }

    fun setButtons(buttons: List<List<ButtonEntity>>?): KeyboardEntity {
        this.buttons = buttons
        return this
    }

    @Keep
    @Serializable
    class ButtonEntity {
        var color: String? = null
            private set

        var type: String? = null
            private set

        var label: String? = null
            private set

        var link: String? = null
            private set

        var payload: String? = null
            private set

        fun setColor(color: String?): ButtonEntity {
            this.color = color
            return this
        }

        fun setType(type: String?): ButtonEntity {
            this.type = type
            return this
        }

        fun setLabel(label: String?): ButtonEntity {
            this.label = label
            return this
        }

        fun setLink(link: String?): ButtonEntity {
            this.link = link
            return this
        }

        fun setPayload(payload: String?): ButtonEntity {
            this.payload = payload
            return this
        }
    }
}